docs/site/BelongsTo-relation.md

```diff
@@ -75,18 +100,51 @@ export interface OrderRelations {
 export type OrderWithRelations = Order & OrderRelations;
 ```
 
-The definition of the `belongsTo` relation is inferred by using the `@belongsTo`
-decorator. The decorator takes in a function resolving the target model class
-constructor and designates the relation type. It also calls `property()` to
-ensure that the decorated property is correctly defined.
+The standard naming convention for the foreign key property in the source model
+is `relation name` + `Id` (for example, Order.customerId).
```

Comment:
The explanation here looks weird. It seems like that we define the relation name first, then we decide the default value of our foreign key name. This is because `keyFrom` defaults to the decorated property name, not `targetModel` + `Id`.

I wanted to point out in this PR that, in our code, the way we get the default relation name is by modifying the decorated property name:
```
customerId -> (get rid of `id`) -> customer
```
I found this code is really confusing. For example, if we decorate the foreign key name as
```ts
@belongsTo(..)
anyRandomNameId
```
the default relation name will be `anyRandomName`, which doesn't make sense at all. I guess it was designed to use the default foreign key `customerId`. But the foreign key doesn't default to `targetModel` + `Id`. It defaults to the decorated name. That's the reason makes it hard to document. In short, the code doesn't fully deal with default values as we expected.
 
We might want to improve the way to get the relation name in `belongTo` in the future.

Reply:
yes the code is not very self explanatory in terms of defining relations as of now

Reply:
Well, the FK usually refers to the target model, such as `customerId`. `anyRandomNameId` does not name the FK meaningfully. In such cases, the options.name should be used.